src/lib.rs

```diff
@@ -587,19 +587,22 @@ pub type DynObjectStore = dyn ObjectStore;
 pub type MultipartId = String;
 
 /// Universal API to multiple object store services.
+///
+/// For more convience methods, check [`ObjectStoreExt`].
+///
+/// # Contract
+/// This trait is meant as a contract between object store implementations
+/// (e.g. providers, wrappers) and the `object_store` crate itself.
```

Comment:
Is another key goal that `ObjectStore` is meant to be the minimal required API? If so perhaps we can add something like

```suggestion
/// This trait is meant as a contract between object store implementations
/// (e.g. providers, wrappers) and the `object_store` crate itself and is 
/// intended to be the minimum API required for an object store.
```